### DISCUSSION TITLE
trigger siren

### DISCUSSION THREAD

**Discussion by yknot7** [tjko/nxgipd#16] [General]
Any way to trigger the siren using command? will be a good feature add for automation.

Thanks

**Comment by Dogora** [comment]
I see no command to do this in the NX-584 protocol, so the answer is no.

There's a way to toggle the chime mode and something called "Start keypad sounder". Those are the only ways to make noise I can find.

**Comment by yknot7** [comment]
After some research there is no way to blast siren from [commands](https://www.drivehq.com/folder/p9084843/1816086207.aspx)

Here is an inefficient workaround if your system is self monitored:

Setup fire panic button

From outside of programming type in * + 8 + Programming Code. You will then be asked for the device number. Type 0 and press #. You will then be asked for the location. Type 23 and #. This is done in segment 1 of location 23. You want to make sure that options 5, 6 and 7 are on. You will know that they are on because you will see the numbers being displayed. If they are not displayed then press the corresponding numbers on the keypad to make them show up. The numbers correspond to the explanations below:
On enables audible keypad panic.
On enables keypad aux 1 (fire panic).
On enables keypad aux 2 (emergency panic).
Press * to move to the next segment. That will save the info you just changed. Then press # to exit
add the following code to the files
api.py between line 78 and 79
elif args.get('cmd') == 'siren': CONTROLLER.siren(int(args.get('partition', 1)))

client.py on line 43
def siren(self, partition=1): r = self._session.get( self._url + '/command', params={'cmd': 'siren', 'partition': partition}) return r.status_code == 200

controller.py on line 264
For fire siren
def siren(self, partition): self._queue.append([0x3E, 0x04, partition])

Or for medical keypad siren
def arm_exit(self, partition): self._queue.append([0x3E, 0x05, partition])

http://yourip:5007/command?cmd=siren&partition=1 will trigger the fire alarm siren or the medical keypad siren depending on what you added to the controller.py!
